include/range/v3/detail/config.hpp

```diff
@@ -18,19 +18,55 @@
 #include <iosfwd>
 
 #ifndef RANGES_ASSERT
+#if !defined(NDEBUG) && defined(__GNUC__) && !defined(__clang__) && __GNUC__ < 5
+#include <cstdio>
+#include <cstdlib>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        namespace detail
+        {
+            template<class = void>
+            [[noreturn]] void assert_failure(char const *file, int line, char const *msg)
+            {
+                std::fprintf(stderr, "%s(%d): %s\n", file, line, msg);
+                std::abort();
+            }
+        }
+    }
+}
+
+#define RANGES_ASSERT(...) ((__VA_ARGS__) ? void() : \
+    ::ranges::detail::assert_failure(__FILE__, __LINE__, "assertion failed: " #__VA_ARGS__))
+
+#else
```

Comment:
`assert` can't be used in constexpr functions in GCC<5; this implementation can.